service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/PaymentServiceTest.java

```diff
@@ -198,6 +224,32 @@ void shouldMakeCorrectPaymentForC2WithoutNoticeWhenCustomerReferenceIsNullOrEmpt
             verify(feeService).getFeesDataForC2(WITHOUT_NOTICE);
         }
 
+        @ParameterizedTest
+        @NullAndEmptySource
+        void shouldMakeCorrectPaymentForC2WithoutNoticeWhenCaseReferenceIsNullOrEmpty(
+            final String clientCode) {
+            CaseData caseData = CaseData.builder()
+                .caseLocalAuthority("LA")
+                .c2DocumentBundle(List.of(element(C2DocumentBundle.builder()
+                    .type(WITHOUT_NOTICE)
+                    .pbaNumber("PBA123")
+                    .clientCode(clientCode)
+                    .fileReference("customerReference")
+                    .build())))
+                .build();
```

Comment:
Looks very similar to your other declaration of CaseData. Extract to a private method called buildCaseData and pass in type as param.

Reply:
yup. Now passing C2ApplicationType enum in the argument